### ISSUE TITLE
Normal dependencies don't go into shrinkwrap with NPM 2.7

### ISSUE THREAD

**Issue by gaearon** [npm/npm#7641]
With NPM 2.5.x, `npm shrinkwrap --dev` put both normal dependencies and devDependencies into shrinkwrap file.

After updating to 2.7.2, it only writes devDependencies. Normal dependencies are missing from shrinkwrap file.

I'm running io 1.5.1.

**Comment by drichard** [comment]
Can confirm this behaviour in 2.7.1. Given that the changelog doesn't mention it, it seems likely to be a bug.

Given this package.json with one production dependency and one dev dependency:

````
{
  "name": "npm-shrinkwrap-test",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "author": "",
  "license": "ISC",
  "dependencies": {
    "jwt-simple": "^0.2.0"
  },
  "devDependencies": {
    "proxyquire": "^1.4.0"
  }
}
````

Running the following commands:

* `npm install`
* `npm shrinkwrap --dev`
* `cat npm-shrinkwrap.json`

````
{
  "name": "npm-shrinkwrap-test",
  "version": "1.0.0",
  "dependencies": {
    "proxyquire": {
      "version": "1.4.0",
      "from": "proxyquire@*",
      "resolved": "http://registry.npmjs.org/proxyquire/-/proxyquire-1.4.0.tgz"
    }
  }
}
````

yields a shrinkwrap file with only the dev dependency in it.

**Comment by othiym23** [comment]
Confirmed in `npm@2.7.2`. I think I know what happened (we landed a patch recently to let you run `npm ls` for both `--prod` and `--dev` trees separately, and the guts of the installer are entangled in unpredictable ways), but not how to fix it. Either way, thanks for bringing up the bug and including the repro case!

**Comment by othiym23** [comment]
I've now released [`npm@2.7.3`](https://github.com/npm/npm/releases/tag/v2.7.3) to address this problem. Thanks a lot for bringing this to our attention!

**Comment by gaearon** [comment]
Thank you!
